### ISSUE TITLE
ereader demo - js execution exceeded timeout

### ISSUE THREAD

**Issue by fabriziogiordano** [cubiq/SwipeView#3]
Hi,

**How to reproduce**
iPhone 4, iOS 4.3.5, WiFi
open http://cubiq.org/dropbox/SwipeView/demo/ereader/ and you get an error:

**Javascript Error on Line 59**
_http://cubiq.org/dropbox/SwipeView/demo/ereader/_
_Javascript execution exceeded timeout._

**Comment by cubiq** [comment]
should be fixed, I also added a progress bar

**Comment by fabriziogiordano** [comment]
Thank you Matteo.

Actually I'd like to share with you an idea about splitting the content.

I was playing with a sort of ereader as well and I started using the -webkit-column-count to split a content horizontally. But after some tests it was not properly supported by Android 2.1.
So I changed and I tested a different way.

This is the idea:
The main page load a div let call it _main_ with the full content inside.
Then I get the _scrollHeight_ of the element, then I set a fixed lineHeight.
As I know the screen height viewport I can calculate how many times I can scroll the content.
If the 3 main divs used for the cache are filled with the same content with clone node we can set the scroll position to move them in relation to the page scrolled.

Sorry probably with some images I can explain better:
http://goo.gl/71w2s

Feel free to edit and add comments if you prefer.

Thanks,
Fabrizio
